### PR TITLE
Gallery 2.1.1

### PR DESCRIPTION
**Pull request by RamZallan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,3 +65,14 @@ an Openshift Origin cluster.
    GALLERY_S3_BUCKET_ID = $s3BucketID
    GALLERY_S3_SECRET_KEY = $s3SecretKey
    ```
+
+## Local Development
+Below are instructions for running gallery locally. It assumes that you have already forked and cloned this repository onto your local machine.
+
+1. Change the line in `__init__.py` that sets the config file from `config.env.py` to `localconfig.env.py`.
+
+2. Get gallery dev secrets from an RTP and fill in `localconfig.env.py`. *DO NOT COMMIT THESE TO GIT*
```
Comment (mxmeinhold) [inline README.md]: Can we add this to the .gitignore?

File: README.md [hunk 2]
```diff
@@ -65,3 +65,14 @@ an Openshift Origin cluster.
...
+
+3. Run `pip install -r requirements.txt`
```
Comment (mxmeinhold) [inline README.md]: This probably should include virtualenv recommendations.

File: localconfig.env.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+# Flask config
+DEBUG=False
+IP=os.environ.get('GALLERY_IP', 'localhost')
+PORT=os.environ.get('GALLERY_PORT', '6969')
+SERVER_NAME = os.environ.get('GALLERY_SERVER_NAME', 'localhost:6969')
+SECRET_KEY = os.environ.get('GALLERY_SECRET_KEY', '')
+
+# LDAP config
+LDAP_URL=os.environ.get('GALLERY_LDAP_URL', 'ldaps://ldap.csh.rit.edu:636')
```
Comment (mxmeinhold) [inline localconfig.env.py]: @devinmatte We don't use ldap.csh anymore right? Pretty sure the csh_ldap package should do auto-discovery.

File: localconfig.env.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+LDAP_USER_OU=os.environ.get('GALLERY_LDAP_USER_OU', 'ou=Users,dc=csh,dc=rit,dc=edu')
+
+# OpenID Connect SSO config
+OIDC_ISSUER = os.environ.get('GALLERY_OIDC_ISSUER', 'https://sso.csh.rit.edu/auth/realms/csh')
+OIDC_CLIENT_ID = os.environ.get('GALLERY_OIDC_CLIENT_ID', 'gallery-dev')
+OIDC_CLIENT_SECRET = os.environ.get('GALLERY_OIDC_CLIENT_SECRET', '')
+
+SQLALCHEMY_DATABASE_URI = os.environ.get(
+    'GALLERY_DATABASE_URI',
+    'postgresql://gallery:undergrowth7803824=exhibitionist@postgres.csh.rit.edu/gallery-dev')
```
Comment (mxmeinhold) [inline localconfig.env.py]: I'm gonna just revoke these.

File: gallery/__init__.py [hunk 5]
```diff
@@ -386,24 +390,25 @@ def add_file(file_name: str, path: str, dir_id: str, description: str, owner: st
...
+    dir_children = [d for d in Directory.query.filter(Directory.parent == dir_model.id).all()]
+    file_children = [f for f in File.query.filter(File.parent == dir_model.id).all()]
+    for file in file_children:
+        if file.thumbnail_uuid != DEFAULT_THUMBNAIL_NAME and not file.hidden:
+            return file.thumbnail_uuid
+    for d in dir_children:
+        if d.thumbnail_uuid != DEFAULT_THUMBNAIL_NAME:
+            return d.thumbnail_uuid
+    # WE HAVE TO GO DEEPER (inception noise)
+    for d in dir_children:
```
Comment (mxmeinhold) [inline gallery/__init__.py]: Should we do this before we check through all the children?

File: README.md (hunk 1 above)
Comment (mxmeinhold) [reply README.md]: Might be good to include a localconfig.env.py with non secret defaults and recommend copying it to something else.

File: gallery/__init__.py (hunk 5 above)
Comment (RamZallan) [reply gallery/__init__.py]: Don't think so, as currently the order it chooses a new thumbnail is:
```
if a child file has valid (non-hidden, non-default) thumbnail, use it
else if a child dir has a valid thumbnail, use it
else recurse through all child dirs to find valid thumbnails within them
```

File: localconfig.env.py (hunk 3 above)
Comment (RamZallan) [reply localconfig.env.py]: Looking at the prod deployment of Gallery, we've removed the `LDAP_URL` and `LDAP_USER_OU` vars, so I'll remove them.

**Pull request merged**